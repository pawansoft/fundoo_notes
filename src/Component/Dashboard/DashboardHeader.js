import AsyncStorage from '@react-native-async-storage/async-storage';
import React, { Component } from 'react';
import { TouchableOpacity } from 'react-native';
import { Appbar, Avatar, Searchbar} from 'react-native-paper';
import Firebase from '../../../config/Firebase';
import dashboardStyle from '../../Style/dashboardStyle';
import UserService from '../../../Services/UserServices/UserService';

export default class DashboardHeader extends Component{
    constructor(props) {
        super(props) 
        this.state ={
            userid: '',
            userDetail: [],
            imageUri: ''
        } 
    }

    componentDidMount = async() => {
       const userId = JSON.parse(await AsyncStorage.getItem('userId'))
       await this.setState({
            userid: userId
       })
       console.log(this.state.userid);

       await UserService._getUserDetailService(this.state.userid).then(async data => {
        let userDetail = data ? data : {}
            await this.setState({
                userDetail: userDetail
            })
       })

        await this.setState({
            imageUri : this.state.userDetail[this.state.userid].photo,
        })
       console.log(this.state.imageUri);
    }

    render(){
        return(
            <Appbar style = {dashboardStyle.headerContainer}>
                <Appbar.Action
                icon = 'menu'
                onPress = {() => this.props.navigation.openDrawer()}/>
                
                <Searchbar
                
                placeholder = 'Search Notes' 
                style = {{width : '60%', color: 'black', backgroundColor: 'grey'}}
                onPress = {this.selectView}/>
                
                <Appbar.Action
                    style = {{marginRight : 10}}
                    icon = {(this.props.listView) ? 'view-grid-outline' : 'view-agenda-outline'}
                    onPress={this.props.onPress}/>
                <TouchableOpacity
                onPress = {this.props.onSelectProfile}>
                <Avatar.Image style = {{flexDirection: "row", justifyContent: "space-between", alignItems: "center"  }} 
                size = {40} 
                source = {(this.state.imageUri == '') ?require('../../assets/Profile.png'): {uri : this.state.imageUri}}/>
                </TouchableOpacity>
            </Appbar>
        )
    }
}