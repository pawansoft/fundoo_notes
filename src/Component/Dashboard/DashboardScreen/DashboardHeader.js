import AsyncStorage from '@react-native-async-storage/async-storage';
import React, { Component } from 'react';
import { TouchableOpacity } from 'react-native';
import { Appbar, Avatar } from 'react-native-paper';
import dashboardStyle from '../../../Style/dashboardStyle';
import UserService from '../../../../Services/UserServices/UserService';

export default class DashboardHeader extends Component {
    constructor(props) {
        super(props)
        this.state = {
            userid: '',
            userDetail: [],
            imageUri: ''
        }
    }

    componentDidMount = async () => {
        const userId = JSON.parse(await AsyncStorage.getItem('userId'))
        await this.setState({
            userid: userId
        })

        await UserService._getUserDetailService(this.state.userid).then(async data => {
            let userDetail = data ? data : {}
            await this.setState({
                userDetail: userDetail
            })
        })

        await this.setState({
            imageUri: this.state.userDetail[this.state.userid].photo,
        })
    }

    handleSearch = () => {
        this.props.navigation.push('Home', { screen: 'Search' })
    }

    render() {
        return (
            <Appbar style={dashboardStyle.headerContainer}>
                <Appbar.Action
                    icon='menu'
                    onPress={() => this.props.navigation.openDrawer()} />

                <Appbar.Content title="Search your note"
                    icon="magnify" onPress={() => this.props.navigation.push('Home', { screen: 'Search' })} />

                <Appbar.Action
                    style={{ marginRight: 10 }}
                    icon={(this.props.listView) ? 'view-grid-outline' : 'view-agenda-outline'}
                    onPress={this.props.onPress} />
                <TouchableOpacity
                    onPress={this.props.onSelectProfile}>
                    <Avatar.Image style={{ flexDirection: "row", justifyContent: "space-between", alignItems: "center" }}
                        size={40}
                        source={(this.state.imageUri == '') ? require('../../../assets/Profile.png') : { uri: this.state.imageUri }} />
                </TouchableOpacity>
            </Appbar>
        )
    }
}