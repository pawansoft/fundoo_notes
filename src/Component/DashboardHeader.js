import React, { Component } from 'react'
import {
    View,
    Text
} from 'react-native';
import { Avatar,  Searchbar } from 'react-native-paper';

export default class DashboardHeader extends Component{
    constructor(props){
        super(props)
        this.state = {
            SearchQuery: ''
        }
    }

    SearchbarHandler = async(SearchQuery) => {
        this.setState({
            SearchQuery: SearchQuery
        })
    }

    render(){
        return(
            <View style = {{flexDirection: 'row', justifyContent: 'flex-end', marginRight: '10%'}}>
                <Text>
                    This is header
                </Text>
                <View>
                <Searchbar placeholder="Search"
                    onChangeText = {this.SearchbarHandler}
                    value = {this.state.SearchQuery} />
                </View>
                <View>
                    <Avatar.Image size = {44} source = {require('../assets/img.png')}/>
                    
                </View>
            </View>
        )
    }
}