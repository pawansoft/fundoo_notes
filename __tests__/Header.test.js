import React from 'react';
import {configure, shallow} from 'enzyme';

import {Appbar, Avatar, Searchbar} from 'react-native-paper';

import Adapter from 'enzyme-adapter-react-16'
import DashboardHeader from '../src/Component/Dashboard/DashboardHeader';

configure({adapter: new Adapter()})

jest.mock('react-native-localization', () => class RNLocalization {
    language = 'en-US'
  
    constructor (props) {
      this.props = props
      this.setLanguage(this.language)
    }
  
    setLanguage (interfaceLanguage) {
      this.language = interfaceLanguage
      if (this.props[interfaceLanguage]) {
        var localizedStrings = this.props[this.language]
        for (var key in localizedStrings) {
          if (localizedStrings.hasOwnProperty(key)) this[key] = localizedStrings[key]
        }
      }
    }
})

describe('test DashboardScreen', () => {
    it('provideSnapshot_whenMatched_testShouldpass', () =>{
        const componnt = shallow(<DashboardHeader/>)
        expect(componnt).toMatchSnapshot();
    })

    it('ProvideHeaderComponent_WhenCountNumberOfAppbarAction_ShouldReturn5', () => {
        const component = shallow(<DashboardHeader />)
        expect(component.find(Appbar.Action)).toHaveLength(2)
    })

    it('test all the icons are rendring in header', () => {
        const component = shallow(<DashboardHeader />)
        expect(component.find(Appbar.Action).at(0).props().icon).toEqual('menu')
        expect(component.find(Searchbar)).toHaveLength(1)
        expect(component.find(Avatar.Image)).toHaveLength(1)   
     })

     it('test list and grid view onpress event', async() => {
        const onPressEvent = jest.fn();
        const component = shallow(<DashboardHeader onPress = {onPressEvent}/>)
        expect(component.instance().state.listView).toBe(true);
        await component.instance().selectView();
        expect(component.instance().state.listView).toBe(false);
     })

})