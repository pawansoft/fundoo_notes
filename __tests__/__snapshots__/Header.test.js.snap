// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test DashboardScreen provideSnapshot_whenMatched_testShouldpass 1`] = `ShallowWrapper {}`;
